ambry-frontend/src/main/java/com/github/ambry/frontend/NamedBlobDeleteHandler.java

```diff
@@ -0,0 +1,109 @@
+ *
+ * In the {@link #handle} method, we would first replace the named blob id in the {@link RequestPath} with the
+ * corresponding regular blob id and then call {@link DeleteBlobHandler#handle} to handle the request.
+ */
+class NamedBlobDeleteHandler {
+  private final AccountAndContainerInjector accountAndContainerInjector;
+  private final FrontendMetrics metrics;
+  private final DeleteBlobHandler deleteBlobHandler;
+  private final NamedBlobDb namedBlobDb;
+
+  /**
+   * Construct a {@link NamedBlobDeleteHandler} for handling requests for deleting blobs in named blob accounts.
+   * @param accountAndContainerInjector The {@link AccountAndContainerInjector} to use.
+   * @param namedBlobDb The {@link NamedBlobDb} to use.
+   * @param metrics The {@link FrontendMetrics} instance where metrics are recorded
+   * @param deleteBlobHandler The {@link DeleteBlobHandler}.
+   */
+  NamedBlobDeleteHandler(AccountAndContainerInjector accountAndContainerInjector, NamedBlobDb namedBlobDb,
+      FrontendMetrics metrics, DeleteBlobHandler deleteBlobHandler) {
+    this.accountAndContainerInjector = accountAndContainerInjector;
+    this.namedBlobDb = namedBlobDb;
+    this.metrics = metrics;
+    this.deleteBlobHandler = deleteBlobHandler;
+  }
+
+  /**
+   * Asynchronously handle named blob delete requests.
+   * @param restRequest the {@link RestRequest} that contains the request parameters and body.
+   * @param restResponseChannel the {@link RestResponseChannel} where headers should be set.
+   * @param callback the {@link Callback} to invoke when the response is ready (or if there is an exception).
+   * @throws RestServiceException
+   */
+  void handle(RestRequest restRequest, RestResponseChannel restResponseChannel, Callback<Void> callback)
+      throws RestServiceException {
+    RequestPath requestPath = getRequestPath(restRequest);
+    // Get blob id and reconstruct RequestPath
+    // this is the three-part named blob id, including account name, container name and the custom blob name.
+    String namedBlobId = requestPath.getOperationOrBlobId(false);
+    NamedBlobPath namedBlobPath = NamedBlobPath.parse(namedBlobId, Collections.emptyMap());
+    // This is a three steps operation
+    // 1. Get regular blob id from named blob id.
+    // 2. Delete regular blob id from Router
+    // 3. Delete database row.
+    // If we failed at step 1, or step 1 returns NotFound, then  we just return the error back to client, indicating failure
+    // If we failed at step 2, client can resend the request to ambry so we can resume the operation. If regular blob is already
+    //    deleted, Router would do a nop and return success
+    // If we failed at step 3, client can still resend the delete request to delete this blob again.
+    namedBlobDb.get(namedBlobPath.getAccountName(), namedBlobPath.getContainerName(), namedBlobPath.getBlobName())
```

Comment:
Would it be better if we call convertId in AmbryIdConverterFactory for all namedBlobPath -> blobId mapping?  And inside the NamedBlobDB::delete, we already handle if namedBlob not found issue, do we still need to get the blob first?

Reply:
+1. Yeah, wondering if it would be better to handle namedBlobPath -> blobId operations using `AmbryIdConverter` to hide the namedBlobDb operations from rest of the classes. Since the request path is replaced once the conversion is done, regular blob delete/get handling convert() methods would act on regular blob ID.

Reply:
I disagree. This is one of the major reason why I am doing this, which is to separate named blob operations into it's own components, so that if one day we don't want named blob operations, we can just remove this file and not worry about id converter since id converter has nothing to do with named blob. 

That's how I see named blob, it's an add-on, extended component, not part of core ambry read/write path. Core ambry should know as little as possible about named blob. 

I will admit this is opinionated. We can talk more about this offline.